PowerShell/ScubaGear/Modules/CreateReport/CreateReport.psm1

```diff
@@ -270,13 +270,33 @@ function New-Report {
         # Create a section header for the licensing information
         $LicensingHTML = "<h2>Tenant Licensing Information</h2>" + $LicenseTable
 
+        # Create a section for privileged service principals
+        $privilegedServicePrincipalsTable = $SettingsExport.privileged_service_principals.psobject.properties | ForEach-Object {
```

Comment:
Logic in AADProvider:

If $PrivilegedServicePrincipals is $null, then it equals {}. Otherwise, set equal to $PrivilegedServicePrincipals
Then in the AADProvider json set "privileged_service_principals" equal to $PrivilegedServicePrincipals

If "privileged_service_principals" = {}, then $principal.DisplayName, $principal.ServicePrincipalId, etc. are nonexistent properties. Can we add some error handling here? 

We could check `$SettingsExport.privileged_service_principals.PSObject.Count > 0` or some other condition.